workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -110,69 +116,106 @@ private Worker(string workerType, Connection connection, ILogDispatcher logDispa
+            ReceptionistConfig config,
             ConnectionParameters connectionParameters,
             ILogDispatcher logger, Vector3 origin)
         {
+            // TODO: Remove when UTY-1578 is fixed.
+#if UNITY_STANDALONE_LINUX
+            connectionParameters.EnableProtocolLoggingAtStartup = false;
+            Debug.LogWarning("Automatically disabling protocol logging on Linux workers to prevent crashes.");
+#endif
             using (var connectionFuture =
-                Connection.ConnectAsync(config.ReceptionistHost, config.ReceptionistPort, config.WorkerId, connectionParameters))
+                Connection.ConnectAsync(config.ReceptionistHost, config.ReceptionistPort, config.WorkerId,
+                    connectionParameters))
             {
                 return await TryToConnectAsync(connectionFuture, connectionParameters.WorkerType, logger, origin);
             }
         }
 
         /// <summary>
-        /// Connects to the SpatialOS Runtime via the Locator service and creates a <see cref="Worker"/> object asynchronously.
+        ///     Connects to the SpatialOS Runtime via the Locator service and creates a <see cref="Worker" /> object
+        ///     asynchronously.
         /// </summary>
-        /// <param name="config">The <see cref="LocatorConfig"/> object stores the configuration needed to connect via the Receptionist Service.</param>
-        /// <param name="connectionParameters">The <see cref="ConnectionParameters"/> storing </param>
+        /// <param name="config">
+        ///     The <see cref="LocatorConfig" /> object stores the configuration needed to connect via the
+        ///     Receptionist Service.
+        /// </param>
+        /// <param name="connectionParameters">The <see cref="ConnectionParameters" /> storing </param>
         /// <param name="logger">The logger used by this worker.</param>
         /// <param name="origin">The origin of this worker in the local Unity space.</param>
-        /// <returns>A <see cref="Task{TResult}"/> to run this method asynchronously and retrieve the created <see cref="Worker"/> object upon connecting successfully.</returns>
+        /// <returns>
+        ///     A <see cref="Task{TResult}" /> to run this method asynchronously and retrieve the created
+        ///     <see cref="Worker" /> object upon connecting successfully.
+        /// </returns>
         public static async Task<Worker> CreateWorkerAsync(
             LocatorConfig parameters,
             ConnectionParameters connectionParameters,
             ILogDispatcher logger, Vector3 origin)
         {
-                using (var locator = new Locator(parameters.LocatorHost, parameters.LocatorParameters))
+            // TODO: Remove when UTY-1578 is fixed.
+#if UNITY_STANDALONE_LINUX
+            connectionParameters.EnableProtocolLoggingAtStartup = false;
+            Debug.LogWarning("Automatically disabling protocol logging on Linux workers to prevent crashes.");
```

Comment:
should we always print that warning? or only if we actually do disable it

Reply:
I actually think it doesn't matter either way in this particular case